# 신규 민간 아파트 분양가

 * 2013년부터 2018년까지 부동산 가격의 변동추세가 아파트 분양가에 반영이 되는지를 알아보고자 함

## 데이터 로드

In [1]:
import pandas as pd

#최근 분양가 파일 로드
df_last=pd.read_csv("C:\\Users\\user7\\Desktop\\data\\data1\\주택도시보증공사_전국 평균 분양가격(2019년 12월).csv",encoding="cp949")

#행,열 갯수 파악
df_last.shape

(4335, 5)

In [2]:
df_last.head()

지역명               규모구분    연도   월 분양가격(㎡)
0  서울                 전체  2015  10    5841
1  서울         전용면적 60㎡이하  2015  10    5652
2  서울   전용면적 60㎡초과 85㎡이하  2015  10    5882
3  서울  전용면적 85㎡초과 102㎡이하  2015  10    5721
4  서울        전용면적 102㎡초과  2015  10    5879

In [3]:
df_last.tail()

지역명               규모구분    연도   월 분양가격(㎡)
4330  제주                 전체  2019  12    3882
4331  제주         전용면적 60㎡이하  2019  12     NaN
4332  제주   전용면적 60㎡초과 85㎡이하  2019  12    3898
4333  제주  전용면적 85㎡초과 102㎡이하  2019  12     NaN
4334  제주        전용면적 102㎡초과  2019  12    3601

In [4]:
#2013-2015년 분양가격 데이터 로드
df_first=pd.read_csv("C:\\Users\\user7\\Desktop\\data\\data1\\전국 평균 분양가격(2013년 9월부터 2015년 8월까지).csv",encoding="cp949")

In [5]:
df_first.shape

(17, 22)

In [6]:
df_first.head()

지역  2013년12월  2014년1월  2014년2월  2014년3월  2014년4월  2014년5월  2014년6월  \
0  서울     18189    17925    17925    18016    18098    19446    18867   
1  부산      8111     8111     9078     8965     9402     9501     9453   
2  대구      8080     8080     8077     8101     8267     8274     8360   
3  인천     10204    10204    10408    10408    10000     9844    10058   
4  광주      6098     7326     7611     7346     7346     7523     7659   

   2014년7월  2014년8월  ...  2014년11월  2014년12월  2015년1월  2015년2월  2015년3월  \
0    18742    19274  ...     20242     20269    20670    20670    19415   
1     9457     9411  ...      9208      9208     9204     9235     9279   
2     8360     8370  ...      8439      8253     8327     8416     8441   
3     9974     9973  ...     10020     10020    10017     9876     9876   
4     7612     7622  ...      7752      7748     7752     7756     7861   

   2015년4월  2015년5월  2015년6월  2015년7월  2015년8월  
0    18842    18367    18374    18152    18443  
1     9327     9345     9515     9559     9581  
2     8446     8568     8542     8542     8795  
3     9938    10551    10443    10443    10449  
4     7914     7877     7881     8089     8231  

[5 rows x 22 columns]

In [7]:
df_first.tail()

지역  2013년12월  2014년1월  2014년2월  2014년3월  2014년4월  2014년5월  2014년6월  \
12  전북      6282     6281     5946     5966     6277     6306     6351   
13  전남      5678     5678     5678     5696     5736     5656     5609   
14  경북      6168     6168     6234     6317     6412     6409     6554   
15  경남      6473     6485     6502     6610     6599     6610     6615   
16  제주      7674     7900     7900     7900     7900     7900     7914   

    2014년7월  2014년8월  ...  2014년11월  2014년12월  2015년1월  2015년2월  2015년3월  \
12     6319     6436  ...      6583      6583     6583     6583     6542   
13     5780     5685  ...      5768      5784     5784     5833     5825   
14     6556     6563  ...      6881      6989     6992     6953     6997   
15     6613     6606  ...      7125      7332     7592     7588     7668   
16     7914     7914  ...      7724      7739     7739     7739     7826   

    2015년4월  2015년5월  2015년6월  2015년7월  2015년8월  
12     6551     6556     6601     6750     6580  
13     5940     6050     6243     6286     6289  
14     7006     6966     6887     7035     7037  
15     7683     7717     7715     7723     7665  
16     7285     7285     7343     7343     7343  

[5 rows x 22 columns]

## 데이터 전처리

In [8]:
df_last.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4335 entries, 0 to 4334
Data columns (total 5 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   지역명      4335 non-null   object
 1   규모구분     4335 non-null   object
 2   연도       4335 non-null   int64 
 3   월        4335 non-null   int64 
 4   분양가격(㎡)  4058 non-null   object
dtypes: int64(2), object(3)
memory usage: 169.5+ KB


* 지역명, 규모구분, 연도,월은 데이터의 개수가 4335개인데 분양가격만 4058개임을 확인함을 통해 결측치의 존재를 파악

### 결측치 확인

In [9]:
df_last.isnull()

지역명   규모구분     연도      월  분양가격(㎡)
0     False  False  False  False    False
1     False  False  False  False    False
2     False  False  False  False    False
3     False  False  False  False    False
4     False  False  False  False    False
...     ...    ...    ...    ...      ...
4330  False  False  False  False    False
4331  False  False  False  False     True
4332  False  False  False  False    False
4333  False  False  False  False     True
4334  False  False  False  False    False

[4335 rows x 5 columns]

In [10]:
#결측치 갯수 구하기

True==1
False==0

df_last.isnull().sum()

지역명          0
규모구분         0
연도           0
월            0
분양가격(㎡)    277
dtype: int64

### 데이터 타입 변경
* 분양가격이 object(문자)타입이므로 수치 데이터로 변경
* .astype(int)는 결측치가 섞인 상태라면 변환이 되지않는다. 따라서 pd.to_numeric을 이용하고 error를 무시하고 강제로 출력함

In [11]:
df_last["분양가격"]=pd.to_numeric(df_last["분양가격(㎡)"],errors='coerce')
df_last["분양가격"]

0       5841.0
1       5652.0
2       5882.0
3       5721.0
4       5879.0
         ...  
4330    3882.0
4331       NaN
4332    3898.0
4333       NaN
4334    3601.0
Name: 분양가격, Length: 4335, dtype: float64

In [12]:
type(pd.np.nan)

<ipython-input-12-e71c757bba12>:1: FutureWarning: The pandas.np module is deprecated and will be removed from pandas in a future version. Import numpy directly instead
  type(pd.np.nan)


float

### 평당 분양 가격 구하기
* 2013-2015년 데이터는 평당 분양 가격 기준이므로 최근데이터의 분양가격에 3.3을 곱한 평당분양가격 컬럼을 추가함

In [13]:
df_last["평당분양가격"]=df_last["분양가격"]*3.3
df_last.head(1)

지역명 규모구분    연도   월 분양가격(㎡)    분양가격   평당분양가격
0  서울   전체  2015  10    5841  5841.0  19275.3

### 분양 가격 요약

In [14]:
df_last.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4335 entries, 0 to 4334
Data columns (total 7 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   지역명      4335 non-null   object 
 1   규모구분     4335 non-null   object 
 2   연도       4335 non-null   int64  
 3   월        4335 non-null   int64  
 4   분양가격(㎡)  4058 non-null   object 
 5   분양가격     3957 non-null   float64
 6   평당분양가격   3957 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 237.2+ KB


In [15]:
df_last["분양가격(㎡)"].describe()

count     4058
unique    1753
top       2221
freq        17
Name: 분양가격(㎡), dtype: object

In [16]:
df_last["분양가격"].describe()

count     3957.000000
mean      3238.128633
std       1264.309933
min       1868.000000
25%       2441.000000
50%       2874.000000
75%       3561.000000
max      12728.000000
Name: 분양가격, dtype: float64

* 중앙값은 2874인데 최댓값이 굉장히 커서 평균값이 중앙값보다 큰 3238 이 나옴

### 데이터 간결화
* 규모구분 칼럼은 전용면적에 관한 데이터이다. 규모구분이란 단어보다 전용면적이 더 직관적이므로 칼럼이름을 전용면적이라고 새로 만들고 기존 전용면적이라는 단어와 초과,이하라는 불필요한 단어는 제거함
* 60초과 85이하는 60~80으로 변경

In [17]:
df_last["규모구분"].unique()

array(['전체', '전용면적 60㎡이하', '전용면적 60㎡초과 85㎡이하', '전용면적 85㎡초과 102㎡이하',
       '전용면적 102㎡초과'], dtype=object)

In [18]:
df_last["규모구분"]

0                      전체
1              전용면적 60㎡이하
2        전용면적 60㎡초과 85㎡이하
3       전용면적 85㎡초과 102㎡이하
4             전용면적 102㎡초과
              ...        
4330                   전체
4331           전용면적 60㎡이하
4332     전용면적 60㎡초과 85㎡이하
4333    전용면적 85㎡초과 102㎡이하
4334          전용면적 102㎡초과
Name: 규모구분, Length: 4335, dtype: object

In [19]:
df_last["전용면적"]=df_last["규모구분"].str.replace("전용면적","")
df_last["전용면적"]=df_last["전용면적"].str.replace("초과","~")
df_last["전용면적"]=df_last["전용면적"].str.replace("이하","")
df_last["전용면적"]=df_last["전용면적"].str.replace(" ","").str.strip() #앞 뒤 공백도 제거(str.strip)
df_last["전용면적"]

0             전체
1            60㎡
2        60㎡~85㎡
3       85㎡~102㎡
4          102㎡~
          ...   
4330          전체
4331         60㎡
4332     60㎡~85㎡
4333    85㎡~102㎡
4334       102㎡~
Name: 전용면적, Length: 4335, dtype: object

### 필요없는 칼럼 제거

* 규모구분, 분양가격(㎡) 은 필요없음

In [20]:
df_last=df_last.drop(["규모구분","분양가격(㎡)"],axis=1) #행은 0, 열은 1

In [21]:
df_last.head(1)

지역명    연도   월    분양가격   평당분양가격 전용면적
0  서울  2015  10  5841.0  19275.3   전체

## 데이터 집계

* 지역명으로 분양가격의 평균을 그리고 막대그래프로 시각화

In [24]:
df_last.groupby(["지역명"])["평당분양가격"].mean()

지역명
강원     7890.750000
경기    13356.895200
경남     9268.778138
경북     8376.536515
광주     9951.535821
대구    11980.895455
대전    10253.333333
부산    12087.121200
서울    23599.976400
세종     9796.516456
울산    10014.902013
인천    11915.320732
전남     7565.316532
전북     7724.235484
제주    11241.276712
충남     8233.651883
충북     7634.655600
Name: 평당분양가격, dtype: float64

In [25]:
df_last.groupby(["지역명"])["평당분양가격"].describe()

count          mean          std      min        25%       50%  \
지역명                                                                   
강원   242.0   7890.750000  1044.359840   6639.6   7165.950   7581.75   
경기   250.0  13356.895200  1969.310711  10160.7  11729.025  13168.65   
경남   247.0   9268.778138  1356.695327   7260.0   8233.500   9187.20   
경북   241.0   8376.536515   850.823275   6949.8   7728.600   8335.80   
광주   201.0   9951.535821  1506.593261   7428.3   9068.400   9527.10   
대구   242.0  11980.895455  2018.395637   8259.9  10399.950  11904.75   
대전   198.0  10253.333333  2245.370020   8002.5   8580.000   9401.70   
부산   250.0  12087.121200  1447.322754   9669.0  10709.325  12338.70   
서울   250.0  23599.976400  3969.953218  16701.3  21318.825  22497.75   
세종   237.0   9796.516456  1096.551246   8487.6   8906.700   9197.10   
울산   149.0  10014.902013   899.543667   7992.6   9484.200  10127.70   
인천   246.0  11915.320732  1642.963530   9537.0  10808.325  11378.40   
전남   248.0   7565.316532   773.929820   6289.8   7098.300   7585.05   
전북   248.0   7724.235484   868.364876   6270.0   7248.450   7659.30   
제주   219.0  11241.276712  2351.630010   7342.5   9929.700  10787.70   
충남   239.0   8233.651883   741.954128   6926.7   7585.050   8134.50   
충북   250.0   7634.655600   796.759331   6164.4   6969.600   7543.80   

           75%      max  
지역명                      
강원    8281.350  11873.4  
경기   14600.025  18711.0  
경남    9698.700  14199.9  
경북    8824.200  11408.1  
광주   10800.900  16107.3  
대구   13330.350  17021.4  
대전   11615.175  16094.1  
부산   12972.300  15255.9  
서울   25072.575  42002.4  
세종   10388.400  12972.3  
울산   10474.200  11860.2  
인천   12480.600  17120.4  
전남    7923.300  10074.9  
전북    8241.750  10071.6  
제주   12553.200  18024.6  
충남    8768.100  10563.3  
충북    8250.000   9421.5

In [26]:
# 전용면적으로 평당분양가격의 평균
df_last.groupby(["전용면적"])["평당분양가격"].mean()

전용면적
102㎡~       11517.705634
60㎡         10375.137421
60㎡~85㎡     10271.040071
85㎡~102㎡    11097.599573
전체          10276.086207
Name: 평당분양가격, dtype: float64

In [30]:
df_last.groupby(["지역명","전용면적"])["평당분양가격"].mean().round()

지역명  전용면적    
강원   102㎡~       8311.0
     60㎡         7567.0
     60㎡~85㎡     7486.0
     85㎡~102㎡    8750.0
     전체          7478.0
                  ...  
충북   102㎡~       8195.0
     60㎡         7103.0
     60㎡~85㎡     7264.0
     85㎡~102㎡    8391.0
     전체          7219.0
Name: 평당분양가격, Length: 85, dtype: float64

In [32]:
df_last.groupby(["전용면적","지역명"])["평당분양가격"].mean().unstack().round() #unstack은 데이터프레임으로 바꿔줌

지역명           강원       경기       경남      경북       광주       대구       대전  \
전용면적                                                                    
102㎡~     8311.0  14772.0  10358.0  9157.0  11042.0  13087.0  14877.0   
60㎡       7567.0  13252.0   8689.0  7883.0   9431.0  11992.0   9176.0   
60㎡~85㎡   7486.0  12524.0   8619.0  8061.0   9911.0  11779.0   9711.0   
85㎡~102㎡  8750.0  13678.0  10018.0  8774.0   9296.0  11141.0   9037.0   
전체        7478.0  12560.0   8659.0  8079.0   9904.0  11771.0   9786.0   

지역명            부산       서울       세종       울산       인천      전남      전북  \
전용면적                                                                    
102㎡~     13208.0  23446.0  10107.0   9974.0  14362.0  8168.0  8194.0   
60㎡       11354.0  23213.0   9324.0   9202.0  11241.0  7210.0  7610.0   
60㎡~85㎡   11865.0  22787.0   9775.0  10503.0  11384.0  7269.0  7271.0   
85㎡~102㎡  12073.0  25944.0   9848.0   8861.0  11528.0  7909.0  8276.0   
전체        11936.0  22610.0   9805.0  10493.0  11257.0  7284.0  7293.0   

지역명            제주      충남      충북  
전용면적                               
102㎡~     10523.0  8689.0  8195.0  
60㎡       14022.0  7911.0  7103.0  
60㎡~85㎡   10621.0  7819.0  7264.0  
85㎡~102㎡  10709.0  9120.0  8391.0  
전체        10785.0  7815.0  7219.0

In [38]:
g=df_last.groupby(["연도","지역명"])["평당분양가격"].mean().unstack().round()
g

지역명       강원       경기       경남      경북       광주       대구       대전       부산  \
연도                                                                           
2015  7188.0  11061.0   8459.0  7464.0   7917.0   9019.0   8191.0  10377.0   
2016  7163.0  11685.0   8497.0  7753.0   9191.0  10282.0   8911.0  10744.0   
2017  7274.0  12305.0   8787.0  8281.0   9614.0  12207.0   9957.0  11561.0   
2018  8219.0  14258.0   9328.0  8681.0   9527.0  12139.0  10234.0  12890.0   
2019  8934.0  15666.0  10698.0  9050.0  12112.0  14082.0  12619.0  13538.0   

지역명        서울       세종       울산       인천      전남      전북       제주      충남  \
연도                                                                          
2015  20316.0   8765.0   9368.0  10976.0  6799.0  7110.0   7951.0  7690.0   
2016  21753.0   8858.0   9583.0  11099.0  6937.0  6907.0   9567.0  7958.0   
2017  21831.0   9133.0  10667.0  11641.0  7373.0  7399.0  12567.0  8198.0   
2018  23202.0  10340.0  10241.0  11882.0  7930.0  8175.0  11936.0  8202.0   
2019  28287.0  11299.0  10216.0  13250.0  8219.0  8532.0  11828.0  8749.0   

지역명       충북  
연도            
2015  6829.0  
2016  7133.0  
2017  7473.0  
2018  8149.0  
2019  7971.0

In [37]:
df_last.groupby(["연도","지역명"])["평당분양가격"].mean().unstack().round().T # .T를 하면 행과 열을 바꿈


연도      2015     2016     2017     2018     2019
지역명                                             
강원    7188.0   7163.0   7274.0   8219.0   8934.0
경기   11061.0  11685.0  12305.0  14258.0  15666.0
경남    8459.0   8497.0   8787.0   9328.0  10698.0
경북    7464.0   7753.0   8281.0   8681.0   9050.0
광주    7917.0   9191.0   9614.0   9527.0  12112.0
대구    9019.0  10282.0  12207.0  12139.0  14082.0
대전    8191.0   8911.0   9957.0  10234.0  12619.0
부산   10377.0  10744.0  11561.0  12890.0  13538.0
서울   20316.0  21753.0  21831.0  23202.0  28287.0
세종    8765.0   8858.0   9133.0  10340.0  11299.0
울산    9368.0   9583.0  10667.0  10241.0  10216.0
인천   10976.0  11099.0  11641.0  11882.0  13250.0
전남    6799.0   6937.0   7373.0   7930.0   8219.0
전북    7110.0   6907.0   7399.0   8175.0   8532.0
제주    7951.0   9567.0  12567.0  11936.0  11828.0
충남    7690.0   7958.0   8198.0   8202.0   8749.0
충북    6829.0   7133.0   7473.0   8149.0   7971.0

* pivot table 사용 

In [39]:
pd.pivot_table(df_last,index=["지역명"],values=["평당분양가격"],aggfunc="mean")

평당분양가격
지역명              
강원    7890.750000
경기   13356.895200
경남    9268.778138
경북    8376.536515
광주    9951.535821
대구   11980.895455
대전   10253.333333
부산   12087.121200
서울   23599.976400
세종    9796.516456
울산   10014.902013
인천   11915.320732
전남    7565.316532
전북    7724.235484
제주   11241.276712
충남    8233.651883
충북    7634.655600

In [40]:
pd.pivot_table(df_last,index=["전용면적"],values=["평당분양가격"],aggfunc="mean")

평당분양가격
전용면적                  
102㎡~     11517.705634
60㎡       10375.137421
60㎡~85㎡   10271.040071
85㎡~102㎡  11097.599573
전체        10276.086207

In [43]:
df_last.pivot_table(index="전용면적",columns="지역명",values="평당분양가격",aggfunc="mean").round()

지역명           강원       경기       경남      경북       광주       대구       대전  \
전용면적                                                                    
102㎡~     8311.0  14772.0  10358.0  9157.0  11042.0  13087.0  14877.0   
60㎡       7567.0  13252.0   8689.0  7883.0   9431.0  11992.0   9176.0   
60㎡~85㎡   7486.0  12524.0   8619.0  8061.0   9911.0  11779.0   9711.0   
85㎡~102㎡  8750.0  13678.0  10018.0  8774.0   9296.0  11141.0   9037.0   
전체        7478.0  12560.0   8659.0  8079.0   9904.0  11771.0   9786.0   

지역명            부산       서울       세종       울산       인천      전남      전북  \
전용면적                                                                    
102㎡~     13208.0  23446.0  10107.0   9974.0  14362.0  8168.0  8194.0   
60㎡       11354.0  23213.0   9324.0   9202.0  11241.0  7210.0  7610.0   
60㎡~85㎡   11865.0  22787.0   9775.0  10503.0  11384.0  7269.0  7271.0   
85㎡~102㎡  12073.0  25944.0   9848.0   8861.0  11528.0  7909.0  8276.0   
전체        11936.0  22610.0   9805.0  10493.0  11257.0  7284.0  7293.0   

지역명            제주      충남      충북  
전용면적                               
102㎡~     10523.0  8689.0  8195.0  
60㎡       14022.0  7911.0  7103.0  
60㎡~85㎡   10621.0  7819.0  7264.0  
85㎡~102㎡  10709.0  9120.0  8391.0  
전체        10785.0  7815.0  7219.0

In [45]:
p=pd.pivot_table(df_last,index=["연도","지역명"],values="평당분양가격").round()
p.loc[2018]

평당분양가격
지역명         
강원    8219.0
경기   14258.0
경남    9328.0
경북    8681.0
광주    9527.0
대구   12139.0
대전   10234.0
부산   12890.0
서울   23202.0
세종   10340.0
울산   10241.0
인천   11882.0
전남    7930.0
전북    8175.0
제주   11936.0
충남    8202.0
충북    8149.0